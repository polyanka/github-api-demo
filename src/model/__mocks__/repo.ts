import { IRepo } from '../repo';
export const repos: IRepo[] = [
  {
    id: 1,
    name: 'name1',
    description: 'description1',
    created_at: '2021-08-30T21:19:21Z',
    html_url: 'https://www.example1.com',
    stargazers_count: 5,
    watchers_count: 3,
    language: 'JavaScript',
  },
  {
    id: 2,
    name: 'name2',
    description: 'description2',
    created_at: '2021-09-30T22:56:01Z',
    html_url: 'https://www.example2.com',
    stargazers_count: 1,
    watchers_count: 12,
    language: 'TypeScript',
  },
];
