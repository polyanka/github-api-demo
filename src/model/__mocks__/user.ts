import { IUser } from '../user';

export const user: IUser = {
  avatar_url: 'https://www.example.com/avatar_url.png',
  id: 1,
  login: 'Login',
  name: 'User',
  blog: 'https://www.example-blog.com',
  location: null,
  public_repos: 0,
  public_gists: 2,
  followers: 2,
  following: 0,
  company: null,
};
