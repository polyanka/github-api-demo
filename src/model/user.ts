export interface IUserBase {
  id: number;
  login: string;
  avatar_url: string;
}

export interface IUsers {
  users: IUserBase[];
}

export interface IUser extends IUserBase {
  name: string;
  blog: string;
  location: string;
  public_repos: number;
  public_gists: number;
  followers: number;
  following: number;
  company: string;
}
