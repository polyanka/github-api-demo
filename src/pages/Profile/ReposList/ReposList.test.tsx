import { repos } from '@src/model/__mocks__/repos';
import { user } from '@src/model/__mocks__/user';
import { shallow } from 'enzyme';
import * as React from 'react';
import * as ReactRedux from 'react-redux';

import { ReposList } from './index';

describe('<ReposList />', () => {
  beforeEach(() => {
    jest.spyOn(ReactRedux, 'useDispatch').mockReturnValue(jest.fn());
  });

  it('start render <ReposList />', () => {
    jest.spyOn(ReactRedux, 'useSelector').mockReturnValue({
      repos: null,
      loading: true,
      error: false,
    });

    const wrapper = shallow(<ReposList user={user} />);

    expect(wrapper.find('Spiner')).toHaveLength(1);
    expect(wrapper.find('ErrorMessage')).toHaveLength(0);
    expect(wrapper.find('RepoCard')).toHaveLength(0);
  });

  it('renders correctly ReposList component with repos data', () => {
    jest.spyOn(ReactRedux, 'useSelector').mockReturnValue({
      repos,
      loading: false,
      error: false,
    });

    const wrapper = shallow(<ReposList user={user} />);

    expect(wrapper.find('Spiner')).toHaveLength(0);
    expect(wrapper.find('ErrorMessage')).toHaveLength(0);
    expect(wrapper.find('RepoCard')).toHaveLength(1);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders ErrorMessage', () => {
    jest.spyOn(ReactRedux, 'useSelector').mockReturnValue({
      repos: null,
      loading: false,
      error: true,
    });

    const wrapper = shallow(<ReposList user={user} />);

    expect(wrapper.find('Spiner')).toHaveLength(0);
    expect(wrapper.find('ErrorMessage')).toHaveLength(1);
    expect(wrapper.find('RepoCard')).toHaveLength(0);
  });

  it('renders message if not fount repo', () => {
    jest.spyOn(ReactRedux, 'useSelector').mockReturnValue({
      repos: [],
      loading: false,
      error: false,
    });

    const wrapper = shallow(<ReposList user={user} />);

    expect(wrapper.find('Spiner')).toHaveLength(0);
    expect(wrapper.find('ErrorMessage')).toHaveLength(0);
    expect(wrapper.find('RepoCard')).toHaveLength(0);
    expect(wrapper.find('h3').text()).toBe(`We couldnâ€™t find any repositories on ${user.login}`);
  });
});
