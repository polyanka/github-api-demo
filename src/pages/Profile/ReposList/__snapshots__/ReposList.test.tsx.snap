// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReposList /> renders correctly ReposList component with repos data 1`] = `
<RepoCard
  changePage={[Function]}
  changeSort={[Function]}
  page={1}
  pages={0}
  repos={
    Array [
      Object {
        "created_at": "2021-08-30T21:19:21Z",
        "description": "description1",
        "html_url": "https://www.example1.com",
        "id": 1,
        "language": "JavaScript",
        "name": "name1",
        "stargazers_count": 5,
        "watchers_count": 3,
      },
      Object {
        "created_at": "2021-09-30T22:56:01Z",
        "description": "description2",
        "html_url": "https://www.example2.com",
        "id": 2,
        "language": "TypeScript",
        "name": "name2",
        "stargazers_count": 1,
        "watchers_count": 12,
      },
    ]
  }
  sort={
    Object {
      "order": "desc",
      "orderBy": "full_name",
    }
  }
/>
`;
